web/modules/custom/asset_distribution/asset_distribution.behat.inc

```diff
@@ -0,0 +1,354 @@
+      }
+    };
+    $this->createAssetDistribution($values);
+  }
+
+  /**
+   * Creates a asset distribution from the given property and field data.
+   *
+   * @param array $values
+   *   An optional associative array of values, keyed by property name.
+   *
+   * @return \Drupal\rdf_entity\Entity\Rdf
+   *   A new asset distribution entity.
+   *
+   * @throws \Exception
+   *   Thrown when a given file is not found.
+   */
+  public function createAssetDistribution(array $values) {
+    // Add image.
+    if (!empty($values['field_ad_file'])) {
+      $filename = $values['field_ad_file'];
+      $files_path = $this->getMinkParameter('files_path');
+      $path = rtrim(realpath($files_path), DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . $filename;
+      if (!is_file($path)) {
+        throw new \Exception("File '$filename' was not found in file path '$files_path'.");
+      }
+      // Copy the file into the public files folder and turn it into a File
+      // entity before linking it to the collection.
+      $uri = 'public://' . $filename;
+      $destination = file_unmanaged_copy($path, $uri);
+      $file = File::create(['uri' => $destination]);
+      $file->save();
+      $values['field_ad_file'] = [$file->id()];
+
+      $this->files[$file->id()] = $file;
+    }
+
+    $values['rid'] = 'asset_distribution';
+    $asset_distribution = Rdf::create($values);
+    $asset_distribution->save();
+    $this->assetDistributions[$asset_distribution->id()] = $asset_distribution;
+
+    return $asset_distribution;
+  }
+
+  /**
+   * Deletes an asset distribution.
+   *
+   * @param string $asset_distribution
+   *   The name of the asset distribution.
+   *
+   * @When (I )delete the :asset_distribution asset distribution
+   */
+  public function deleteAssetDistribution($asset_distribution) {
+    /** @var \Drupal\rdf_entity\Entity\Rdf $asset_distribution */
+    $this->getAssetDistributionByName($asset_distribution)->delete();
+  }
+
+  /**
+   * Returns the asset distribution with the given name.
+   *
+   * If multiple asset distributions have the same name,
+   * the first one will be returned.
+   *
+   * @param string $name
+   *   The asset distribution name.
+   *
+   * @return \Drupal\rdf_entity\Entity\Rdf
+   *   The asset distribution.
+   *
+   * @throws \InvalidArgumentException
+   *   Thrown when a asset distribution with the given name does not exist.
+   */
+  protected function getAssetDistributionByName($name) {
+    $query = \Drupal::entityQuery('rdf_entity')
+      ->condition('?entity', 'rdf:type', SparqlArg::uri('http://www.w3.org/ns/adms#AssetDistribution'))
+      ->condition('?entity', SparqlArg::uri('http://purl.org/dc/terms/title'), SparqlArg::literal($name))
+      ->range(0, 1);
+    $result = $query->execute();
+
+    if (empty($result)) {
+      throw new \InvalidArgumentException("The asset distribution with the name '$name' was not found.");
+    }
+
+    return Rdf::load(reset($result));
+  }
```

Comment:
This is duplicated from CollectionSubContext. Let's put it in a trait that reuses code related to RDF entities.